include/package.mk

```diff
@@ -30,7 +29,15 @@ ifdef CONFIG_USE_MIPS16
     TARGET_CFLAGS += -mips16 -minterlink-mips16
   endif
 endif
-ifeq ($(strip $(PKG_IREMAP)),1)
+
+PKG_BUILD_FLAGS?=
+
+# $1=flagname, $2=default (0/1)
+define pkg_build_flag
+$(if $(filter no-$(1),$(PKG_BUILD_FLAGS)),0,$(if $(filter $(1),$(PKG_BUILD_FLAGS)),1,$(2)))
+endef
+
+ifeq ($(call pkg_build_flag,iremap,1),1)
```

Comment:
I'm ok with the changes... Only i think having this amount of freedom may introduce some ""problems"" in the future...

Also to me something like

```
ifeq ($(filter no-iremap, $(PKG_BUILD_fLAGS)),)
```

Looks more direct and clear than a function and a default value...

The idea of these particular flag is to deviate from default configuration in case of problems and ideally they should be there only to workaround unfixable problems.

Ideally each package should use the same flags and if possible apply some fix to follow that.

Giving the possibility to support both no- and normal option CAN result in giving more flexibility and spending less time in trying to make the package more supported...

So IMHO providing only the specific workaround is better than giving the option to follow that or not.

In short I would ask you to drop the define small function and just convert to PKG_BUILD_FLAGS providing directly the supported option.

Free to use whatever syntax you prefer to make more clear what the ifeq does
Example (for the same solution)

```
ifneq ($(filter no-iremap, $(PKG_BUILD_fLAGS)),no-iremap)
```

Reply:
For this specific flag, note that you get the very same result with `PKG_IREMAP:=0` right now.

As for the function, I explicitly wanted a way to:
* let a package set a flag even if the global default doesn't set it
* let a package unset a flag even if the global default sets it

For some flags the default is set, for others unset. Some defaults can be toggled via .config. The function covers all permutations.
If I just check only for $flag and not no-$flag (or the other way around) it doesn't work anymore as intended.
So you really want me to unfold that function for every flag?